cloud-sql/mysql/servlet/src/main/java/com/example/cloudsql/ConnectorConnectionPoolFactory.java

```diff
@@ -0,0 +1,89 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.cloudsql;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import javax.sql.DataSource;
+
+
+public class ConnectorConnectionPoolFactory extends ConnectionPoolFactory {
+
+  // Saving credentials in environment variables is convenient, but not secure - consider a more
+  // secure solution such as https://cloud.google.com/kms/ to help keep secrets safe.
+  private static final String INSTANCE_CONNECTION_NAME =
+      System.getenv("INSTANCE_CONNECTION_NAME");
+  private static final String INSTANCE_UNIX_SOCKET = System.getenv("INSTANCE_UNIX_SOCKET");
+  private static final String DB_USER = System.getenv("DB_USER");
+  private static final String DB_PASS = System.getenv("DB_PASS");
+  private static final String DB_NAME = System.getenv("DB_NAME");
+
+  @SuppressFBWarnings(
+      value = "USBR_UNNECESSARY_STORE_BEFORE_RETURN",
+      justification = "Necessary for sample region tag.")
+  public static DataSource createConnectionPool() {
+    // [START cloud_sql_mysql_servlet_connect_connector]
+    // [START cloud_sql_mysql_servlet_connect_unix]
+    // The configuration object specifies behaviors for the connection pool.
+    HikariConfig config = new HikariConfig();
+
+    // The following URL is equivalent to setting the config options below:
+    // jdbc:mysql:///<DB_NAME>?cloudSqlInstance=<INSTANCE_CONNECTION_NAME>&
+    // socketFactory=com.google.cloud.sql.mysql.SocketFactory&user=<DB_USER>&password=<DB_PASS>
+    // See the link below for more info on building a JDBC URL for the Cloud SQL JDBC Socket Factory
+    // https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory#creating-the-jdbc-url
+
+    // Configure which instance and what database user to connect with.
+    config.setJdbcUrl(String.format("jdbc:mysql:///%s", DB_NAME));
+    config.setUsername(DB_USER); // e.g. "root", "mysql"
+    config.setPassword(DB_PASS); // e.g. "my-password"
+
+    config.addDataSourceProperty("socketFactory", "com.google.cloud.sql.mysql.SocketFactory");
+    config.addDataSourceProperty("cloudSqlInstance", INSTANCE_CONNECTION_NAME);
+
+    // [END cloud_sql_mysql_servlet_connect_connector]
+    // Unix sockets are not natively supported in Java, so it is necessary to use the Cloud SQL JDBC
+    // Socket Factory to connect.
+    // Note: For Java users, the Cloud SQL JDBC Socket Factory can provide authenticated connections
+    // which is preferred to using the Cloud SQL Proxy with Unix sockets.
+    // See https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory for details.
+    if (INSTANCE_UNIX_SOCKET != null) {
+      config.addDataSourceProperty("unixSocketPath", INSTANCE_UNIX_SOCKET);
```

Comment:
How does this work relative to the connector? I would expect to see Unix sockets either work without the connector or not be included here.

Reply:
If we support Unix sockets independently of the connector, I think we should have a UnixConnectionPoolFactory for better region tags.

Reply:
It isn't supported independently of the connector. Java doesn't support unix sockets natively, so it makes sense to include this with the connector sample. See:
https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/2de5b9e3e3c4b31482462be28926d787c4a6cff8/core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java#L235-L247